### PR TITLE
Soft Indexing sequences can now being at non-zero indices

### PR DESCRIPTION
**Pull request by julianxcarter**
This PR addresses another issue found in [PR #106](https://github.com/FHIR/GoFSH/pull/106) on GoFSH. When unwinding soft indexing, Sushi currently makes the assumption that the first element being assigned by an array utilizing soft indexing will always be at index 0. In the event that sushi first encounters an array as it references a non-zero element, this could result in something like the following... 
```
* effective[x] ^constraint[1].key = "us-core-1"
* effective[x] ^constraint[=].severity = #error
* effective[x] ^constraint[=].human = "Datetime must be at least to day."
```
resolving to this:
```
* effective[x] ^constraint[1].key = "us-core-1"
* effective[x] ^constraint[0].severity = #error
* effective[x] ^constraint[0].human = "Datetime must be at least to day."
```
This PR fixes this bug, taking numeric indices into account at all times. The issue can be illustrated by running the unit test included in commit `6035ded`, which should fail in that commit but pass in commit `037cd63` which adds the fix.

### CODE REVIEW

**Review by jafeltra — APPROVED** [review]
These changes and the test make sense to me!

**Comment by cmoesel** [comment]
This seems reasonable to me, but I'm going to run a full regression to ensure it doesn't affect any existing IGs (or if it does, that it affects them in a good way).

**Comment by cmoesel** [comment]
This did well in regression.  In fact, it fixed a (previously undetected) bug in the mCODE IG generation because mCODE uses a numeric index followed by a `[=]` here: https://github.com/HL7/fhir-mCODE-ig/blob/master/input/fsh/EX_Bundle.fsh#L21-L22

So... way to go!

That said, @ngfreiter has asked that we not merge this until he has a chance to review and test it in combination w/ the GoFSH PR that raised this issue.

**Review by cmoesel — APPROVED** [review]
Approve, pending @ngfreiter's additional review.

**Review by ngfreiter — APPROVED** [review]
Sorry for the delay, I've confirmed that when running this with https://github.com/FHIR/GoFSH/pull/106, everything is working as we would want, so awesome!

**Pull request merged**
